src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const stateHistory = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        newState = { ...newState, ...actions[i].extraData };
+        stateHistory.push({ ...newState });
```

Comment:
Instead of pushing the new state inside each case, you can do it once after the switch statement to adhere to the DRY principle.